File: 2021/Day_2-2.py
#Day 2 Part 2

f=0
d=0
a=1000
aim=0
for i in range(a):
    l=input()
    if 'forward' in l:
        f+=int(l[8])
        d+=aim*int(l[8])
    if 'up' in l:
        aim-=int(l[3])
    
    if 'down' in l:
        aim+=int(l[5])
    
print(f*d)
#Bhav Beri

File: 2021/C++/Day_2-2.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    long long hori = 0;
    long long dep = 0;
    long long aim = 0;
    for (int i = 0; i < 1000;i++) {
        string s;
        cin >> s;
        long long n;
        cin >> n;
        if (s == "forward") {
            hori += n;
            dep += (aim * n);
        }
        else if (s == "down")
            aim += n;
        else if (s == "up")
            aim -= n;
    }
    cout << hori * dep;
}
